### ISSUE TITLE
FxA Integration

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-it/ctms-api#326]
┆Issue is synchronized with this [Jira Epic](https://mozilla-hub.atlassian.net/browse/CTMS-89)

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status 9/17 - 

  * In sprint 0 to prep for sprinting. Established team, team artifacts (team agreements, DoR, and DoD), and set up Jira.
  * Created stories and refined them with the team. There are two research spikes that need to be completed before work can start.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status 10/1 -

  * Still need to continue research. This week, team had to focus on OKRs and fixing production issues so that took up time
  * Goal is for team to complete their research spike end of day today.
  * Will meet up on Monday for planning to see if we can sprint.
  * FxA team is working on getting stripe data to be present in Firestore. Ben is working on the PR for that and will have it ready by Monday.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status 10-5-21

  * Started sprinting today. Completed research and planning for sprint 0 today. 
  * Sprint Goal for sprint 1:
    * Revise schema to store subscription status
    * Receive subscription status changes into CTMS

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status 10/18/21 -

  * Impediments this sprint:
    * Out sick toward the end of the sprint
    * Had to work on operations work such as taskcluster support

  * Apple IAP to FxA
    * FxA team will no longer have this done by November. Since we are only having stripe data, we can forget about the apple data for now. 

Reminder that it takes about ~ 3 sprints to forecast and get velocity. This is our first sprint in our project.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Sprint Closure Report - Sprint 1

  * Sprint 1 ends Monday, 10/18.  Closed out one story - CTMS-75 ( https://mozilla-hub.atlassian.net/browse/CTMS-75 ) . 
  * Carryover:
    * CTMS-74: This is close to being done. Will complete in sprint 2
    * CTMS-77: This ended up being a bigger story than we had estimated. This still needs a few more days. Will complete in sprint 2

  * Reasons for carryover:
    * Unplanned support work
    * Unplanned sick days

Sprint 2 Goals

  * Will complete unfinished stories from sprint 1 
  * Will complete 90% of the functionality which covers the number one use case for marketing (email to VPN users). The “provide subscriptions status changes from Firestore” is a bit risky to take in for sprint 2. It will be a stretch goal.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

* Status update 10-19-21
    * Sprint 2 starts today. 
    * If we were to complete all stories for this release, the target date would be mid-December. This brings a timeline and resources risk, due to holidays and time off in December. 
    * We now have a new target date of completion of mid November. We will release once we are done with all the “high priority” MVP stories. We may end up deferring some work (the lower priorities) in order to get the customer some functionality sooner. It would cover 90% of the functionality that marketing is requesting.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status update 11/4/21 -

  * Sprint 3 started this week. Goal is to complete all remaining high level stories by this Friday and move to staging next week. We weren’t able to complete those stories in the last sprint. Friday is a stretch goal; we’re not confident that we can make it by this Friday due to also having a dependency on the infrastructure changes and FxA team. Therefore, we are changing the weekly status to “at risk”
  * Conducted our first retrospective. The team has been great working together and being flexible with the project shifts. The customer has been so collaborate and understanding as well. We added our top three retrospective action items to the backlog. 
  * Stephen received approval from our customer that we can deliver early Dec. If we cannot make it mid-November, we have to wait until early December due to not wanting a production promote during the Thanksgiving holiday.

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status update 11/5/21 -

  * Risk of meeting mid Nov production promote. Dependency on FxA team to complete ctms-90. Waiting to hear Jbuck’s estimates 
  * We won’t be able to complete all stories by today. Will need to push to early next week before we can focus on staging/sandbox environment

**Comment by data-sync-user** [comment]
➤ Rosanna Fraser commented:

Status update 11/30/21 - 

  * Changed status to “on track”
  * Completed sprint 4 yesterday with 4 story points. The majority of the carried over stories were already in progress.
  * Big accomplishments are that devs made the fixes for the design issue, new scenario of VPN with existing users defect, and dashboard updates. Team also did a demo with the stakeholders and they accepted the product. 
  * New sprint goals for sprint 5 are to update CTMS alerts, do more integration testing, run regression test, and improve project documentation. 
  * Target to launch on Dec 6. 
  * Team coordinated with SREs on CTMS and FxA side to get ready for Dec 6 launch date.
